app/serializers/credentials_serializer.rb

```diff
@@ -28,12 +27,21 @@ def serialized_user
         full_address: user.address,
         postcode: user.postcode,
       },
-      organisation_uids: user.organisations.pluck(:uid),
+      organisations: serialized_organisations,
       uid: user.uid,
     }
   end
 
-  def serialized_roles
-    user.role_names_for(application: application)
+  def serialized_organisations
+    user.memberships.map do |membership|
+      organisation = membership.organisation
+
+      {
+          uid: organisation.uid,
+          name: organisation.name,
+          type: organisation.organisation_type,
+          roles: membership.roles
```

Comment:
We need to filter the roles by the application - we don't want to be sending role names to an application that does not support them.

Something like:
```
roles = membership.roles & application.available_role_names
```